resharper/src/main/java/org/sonarlint/intellij/resharper/CFamilyFileExclusionContributor.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2024 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.resharper;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiManager;
+import com.jetbrains.rider.cpp.fileType.psi.CppFile;
+import org.sonarlint.intellij.common.analysis.ExcludeResult;
+import org.sonarlint.intellij.common.analysis.FileExclusionContributor;
+
+import static org.sonarlint.intellij.common.util.SonarLintUtils.isCLion;
+
+public class CFamilyFileExclusionContributor implements FileExclusionContributor {
+
+  @Override
+  public ExcludeResult shouldExclude(Module module, VirtualFile fileToAnalyze) {
+    if (!isCLion()) {
+      return ExcludeResult.notExcluded();
+    }
```

Comment:
Is this check needed? Won't this extension point be loaded only when it is the case?

Reply:
It's loaded when `com.intellij.modules.rider.cpp.core` is present, which can be the case in Rider, or in CLion with ReSharper on. But we don't want to run the 'ReSharper CLion' analysis on Rider for instance

Reply:
Could you add a comment? I think it would be useful to remember why this condition exists